### ISSUE TITLE
Play with Kodi button in Player always prompts to install Kore

### ISSUE THREAD

**Issue by lukashass** [TeamNewPipe/NewPipe#5599]
### Checklist

- [x] I am using the latest version - 0.20.10 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1. Go to any video that supports being played in Kodi
2. Start playing the video
3. Press on the _Play with Kodi_ button in the Player

### Actual behaviour

The prompt "Install missing Kore app?" appears, even though Kore is installed.
(Other _Play with Kodi_ buttons work as intended)

### Expected behavior

Kore should be opened as is the case when any other _Play with Kodi_ button is pressed.
(I tested the buttons in video details and list context menu)

### Device info

 - Android version/Custom ROM version: 11
 - Device model:

**Comment by lukashass** [comment]
Throwing unknown errors might be helpful here.

https://github.com/TeamNewPipe/NewPipe/blob/5bc0a8fba198ae51ce5aaf4940b4a77fed5172aa/app/src/main/java/org/schabi/newpipe/player/Player.java#L3572-L3579

```diff
            try {
                NavigationHelper.playWithKore(context, Uri.parse(getVideoUrl()));
            } catch (final Exception e) {
                if (DEBUG) {
                    Log.i(TAG, "Failed to start kore", e);
                }
-               KoreUtil.showInstallKoreDialog(getParentActivity());
+               if (e === "kore not installed error") {
+                   KoreUtil.showInstallKoreDialog(getParentActivity());
+               } else {
+                   throw e;
+               }
            }
```

**Comment by digitalkram** [comment]
I face the same symptoms (Android 10 however).
 
When reproducing that on a debug build `e.detailMessage`  on `Player.java:3576` is:
```
Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
```

**Comment by digitalkram** [comment]
Here:
https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java#L593
(how to include the code from repo in markdown?)

Doing that:
```diff
      intent.setData(videoURL);
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
      context.startActivity(intent);
```

makes it work again. But I am not a developer so I can't judge if that is a real fix or comes at some cost. I just did what the error suggested. 😉

**Comment by lukashass** [comment]
@digitalkram To include the code in markdown have a look here: https://docs.github.com/en/github/managing-your-work-on-github/creating-a-permanent-link-to-a-code-snippet

**Comment by digitalkram** [comment]
Thanks @lukashass, updated my post based on that

**Comment by TiA4f8R** [comment]
@digitalkram I applied your fix in my PR: #5523. Does the APK from CI works for you?

**Comment by digitalkram** [comment]
Yes, the CI artifact apk works as expected 👍

**Comment by lameventanas** [comment]
Just installed Newpipe 0.21.2 and sadly this fix that is almost 2 months old didn't make it there.
When will we be able to get this back working in a released version?

**Comment by TiA4f8R** [comment]
When #5523 is merged. I hope that my PR can be merged the fastest possible, so you can get the fix for 0.21.3 but I need to do some work before on it.

**Comment by BalooRJ1** [comment]
Hello, not sure if this is on track to be fixed in the next version but I am still seeing this bug in 0.21.3 of Newpipe with Kore installed from F-Droid. Using GrapheneOS on a Pixel 3a.
